eth/plasma.go

```diff
@@ -187,30 +208,39 @@ func (plasma *Plasma) HasTxBeenExited(plasmaBlock *big.Int, position plasmaTypes
 
 	priority := position.Priority()
 	if position.IsDeposit() {
-		e, err = plasma.contract.DepositExits(nil, priority)
+		e, err = plasma.DepositExits(nil, priority)
 	} else {
-		e, err = plasma.contract.TxExits(nil, priority)
+		e, err = plasma.TxExits(nil, priority)
 	}
 
 	// censor spends until the error is fixed
 	if err != nil {
-		plasma.logger.Error(fmt.Sprintf("failed to retrieve exit information about position %s { %s }", position, err))
-		return true
+		logger.Error(fmt.Sprintf("failed to retrieve exit information about position %s { %s }", position, err))
+		return true, err
 	}
 
-	ethBlockNum, err := plasma.ethBlockPeg(plasmaBlock)
-	if err != nil {
-		plasma.logger.Error(fmt.Sprintf("could not get associated ETH Block for plasma block %d: %s", plasmaBlock.Int64(), err.Error()))
-		return true
+	// `Pending` or `Challenged` stateee
+	exited := e.State == 1 || e.State == 3
+
+	// synchronize with the correct ethereum state
+	if plasmaBlockHeight != nil {
```

Comment:
when is this nil?

Reply:
if it's nil, disregards the ethereum block peg and checks against the latest state. This is used in plasmacli